omod/src/main/java/org/openmrs/module/pihcore/page/controller/account/MyAccountPageController.java

```diff
@@ -3,122 +3,83 @@
-                       @RequestParam(value = "phoneNumber", required = false) String phoneNumber,
-                       @RequestParam(value = "defaultLocale", required = false) Locale defaultLocale,
-                       @SpringBean("pihCoreService") PihCoreService pihCoreService,
+    public String post(@MethodParam("getAccount") @BindParams PihAccountDomainWrapper account, BindingResult errors,
                        @SpringBean("messageSource") MessageSource messageSource,
                        @SpringBean("messageSourceService") MessageSourceService messageSourceService,
                        @SpringBean("accountService") AccountService accountService,
-                       @SpringBean("userService") UserService userService,
-                       @SpringBean("personService") PersonService personService,
                        @SpringBean("adminService") AdministrationService administrationService,
-                       @SpringBean("accountValidator") AccountValidator accountValidator, PageModel model,
+                       @SpringBean("pihAccountValidator") PihAccountValidator pihAccountValidator,
+                       PageModel model,
                        HttpServletRequest request) {
 
-        User user = Context.getAuthenticatedUser();
-        Person person = personService.getPerson(user.getPerson().getPersonId());
-        PihAccountDomainWrapper account = pihCoreService.newPihAccountDomainWrapper(person);
-
-        Errors errors = new BeanPropertyBindingResult(account, "account");
-
-        // If username has changed, make sure it isn't already in use
-        if (!account.getUsername().equalsIgnoreCase(username)) {
-            User existingUser = userService.getUserByUsername(username);
-            if (existingUser != null) {
-                errors.rejectValue("username", "emr.user.duplicateUsername");
-            }
-        }
-
-        // If email has changed, make sure it isn't already in use or invalid
-        if (StringUtils.isNotBlank(email) && !email.equalsIgnoreCase(account.getEmail())) {
-            if (!EmailValidator.getInstance().isValid(email)) {
-                errors.rejectValue("email", "error.email.invalid");
-            }
-            else {
-                User existingUser = userService.getUserByUsernameOrEmail(email);
-                if (existingUser != null && !existingUser.equals(account.getUser())) {
-                    errors.rejectValue("email", "emr.account.error.emailAlreadyInUse");
-                }
-            }
-        }
-
-        account.setUsername(username);
-        account.setGivenName(givenName);
-        account.setFamilyName(familyName);
-        account.setGender(gender);
-        account.setEmail(email);
-        account.setPhoneNumber(phoneNumber);
-        account.setDefaultLocale(defaultLocale);
-
-        accountValidator.validate(account, errors);
+        pihAccountValidator.validate(account, errors);
 
```

Comment:
All of the above changes are just to make this controller work the same way as the AccountPageController (I had hacked it in this other way previously trying to fix another issue, but it didn't end up being the cause)